Make openshift-priv/kubernetes stuff visible to config-updater

petr-muller opened the pull request:
Fixes: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9071/pull-ci-openshift-release-master-correctly-sharded-config/19276

These bits are not covered by any automation but being special snowflakes I think it's fine to just enable them manually once.

/cc @droslean

stevekuznetsov commented on core-services/prow/02_config/_plugins.yaml:
```diff
@@ -951,7 +960,15 @@ config_updater:
       gzip: true
       name: job-config-misc
-    ci-operator/jobs/openshift/kubernetes/*.yaml:
+    ci-operator/jobs/openshift-priv/kubernetes/openshift-priv-kubernetes-*.yaml:
```
should be build01 too

petr-muller replied:
Hm, no other `ci-operator/jobs/...` one is?

petr-muller replied:
Do we still need jobs on api.ci?

petr-muller replied:
> Do we still need jobs on api.ci?

Looks like we do, as there are periodic jobs that use them. Which means `build01` will need them too as soon as we migrate one of those jobs there.

stevekuznetsov commented on core-services/prow/02_config/_plugins.yaml:
```diff
@@ -634,7 +634,16 @@ config_updater:
-    ci-operator/config/openshift/kubernetes/*.yaml:
+    ci-operator/config/openshift-priv/kubernetes/openshift-priv-kubernetes*.yaml:
+      clusters:
+        api.ci:
```
we don't need ci-operator config on anything but `app,ci`

petr-muller replied:
OK, I just copied existing stanzas - I'll make a followup PR with a full cleanup?

petr-muller commented:
> Doesn't this make openshift/kubernetes stuff invisible?

I don't think so - the patters for openshift/kubernetes paths were also adjusted: previously they were `openshift/kubernetes/*.yaml` which I changed to `openshift/kubernetes/openshift-kubernetes-*.yaml`. The presubmits fail on any files that are not covered by any pattern, and also on possible filename duplicates.
